apis/discoveryengine/v1alpha1/engine_types.go

```diff
@@ -0,0 +1,142 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	refs "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/k8s/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var DiscoveryEngineEngineGVK = GroupVersion.WithKind("DiscoveryEngineEngine")
+
+// DiscoveryEngineEngineSpec defines the desired state of DiscoveryEngineEngine
+// +kcc:proto=google.cloud.discoveryengine.v1.Engine
+type DiscoveryEngineEngineSpec struct {
+	// Required. The display name of the engine. Should be human readable. UTF-8
+	// encoded string with limit of 1024 characters.
+	// +required
+	DisplayName *string `json:"displayName,omitempty"`
+
+	// The data stores associated with this engine.
+	// For SOLUTION_TYPE_SEARCH and SOLUTION_TYPE_RECOMMENDATION type of
+	// engines, they can only associate with at most one data store.
+	// If solution_type is SOLUTION_TYPE_CHAT, multiple DataStores in the
+	// same Collection can be associated here.
+	// Note that when used in CreateEngineRequest, one DataStore must be
+	// provided as the system will use it for necessary initializations.
+	DataStoreRefs []*DiscoveryEngineDataStoreRef `json:"dataStoreRefs,omitempty"`
+
+	// Required. The solutions of the engine.
+	// +required
+	SolutionType *string `json:"solutionType,omitempty"`
+
+	// The industry vertical that the engine registers.
+	// The restriction of the Engine industry vertical is based on
+	// DataStore: If unspecified, default to `GENERIC`. Vertical on Engine
+	// has to match vertical of the DataStore linked to the engine.
+	IndustryVertical *string `json:"industryVertical,omitempty"`
+
+	// Common config spec that specifies the metadata of the engine.
+	CommonConfig *Engine_CommonConfig `json:"commonConfig,omitempty"`
+
+	/*NOTYET
+	// Optional. Whether to disable analytics for searches performed on this
+	// engine.
+	DisableAnalytics *bool `json:"disableAnalytics,omitempty"`
+	*/
+
+	// Configurations for the Chat Engine. Only applicable if
+	// solution_type is SOLUTION_TYPE_CHAT.
+	ChatEngineConfig *Engine_ChatEngineConfig `json:"chatEngineConfig,omitempty"`
+
+	// Configurations for the Search Engine. Only applicable if
+	// solution_type is SOLUTION_TYPE_SEARCH.
+	SearchEngineConfig *Engine_SearchEngineConfig `json:"searchEngineConfig,omitempty"`
+
+	// Optional. Whether to disable analytics for searches performed on this
+	//  engine.
+	DisableAnalytics *bool `json:"disableAnalytics,omitempty"`
+
+	/* Immutable. The Project that this resource belongs to. */
+	ProjectRef *refs.ProjectRef `json:"projectRef"`
+
+	/* Immutable. Location of the resource. */
+	// +required
+	Location string `json:"location"`
+
+	// Immutable. The collection for the Engine.
+	// +required
+	Collection string `json:"collection"`
```

Comment:
This really looks like a resource reference. But it is not. Do you know what happens?